data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java

```diff
@@ -72,37 +66,35 @@ public DataController(final DataService dataService,
      */
-    @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/octet-stream")
-    public ResponseEntity<StreamingResponseBody> readChunked(@RequestBody final DataRequest dataRequest) {
+    @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<StreamingResponseBody> readChunked(
+            @RequestBody final DataRequest dataRequest) {
         LOGGER.info("Invoking read (chunked): {}", dataRequest);
-        DataReaderRequest readerRequest = dataService.authoriseRequest(dataRequest)
-                .thenApply(maybeReadRequest -> maybeReadRequest
-                        .orElseThrow(() -> new ForbiddenException(
-                                String.format("The token '%s' is not authorised to access the leafResource '%s'", dataRequest.getToken(), dataRequest.getLeafResourceId()))))
-                .join();
 
-        StreamingResponseBody stream = (OutputStream outputStream) -> {
-            Pair<AtomicLong, AtomicLong> recordsAudit = dataService.read(readerRequest, outputStream);
-            AuditSuccessMessage successMessage = auditService.createSuccessMessage(dataRequest, readerRequest, recordsAudit.getFirst().get(), recordsAudit.getSecond().get());
-            auditService.auditSuccess(dataRequest.getToken(), successMessage);
-        };
+        HttpStatus httpStatus = HttpStatus.ACCEPTED;
+        StreamingResponseBody stream = null;
 
-        LOGGER.info("Streaming response: {}", stream);
-        return new ResponseEntity<>(stream, HttpStatus.OK);
-    }
+        //first with the client information about the request, retrieve the authorised resource information
+        AuditableAuthorisedDataRequest auditableAuthorisedDataRequest = auditableDataService.authoriseRequest(dataRequest).join();
+        AuditErrorMessage firstErrorMessage = auditableAuthorisedDataRequest.getAuditErrorMessage();
 
-    /**
-     * This method will add a {@link uk.gov.gchq.palisade.data.serialise.Serialiser} to the
-     * {@link DataService} using the details provided in a yaml file.
-     */
-    @EventListener(ApplicationReadyEvent.class)
-    public void initPostConstruct() {
-        // Add serialiser to the data-service
-        LOGGER.info("Prepopulating using serialiser config: {}", serialiserConfig.getClass());
-        serialiserConfig.getSerialisers().stream()
-                .map(StdSerialiserPrepopulationFactory::build)
-                .peek(entry -> LOGGER.debug(entry.toString()))
-                .forEach(entry -> dataService.addSerialiser(entry.getKey(), entry.getValue()));
-    }
+        if (firstErrorMessage != null) {
+            LOGGER.error("Error occurred processing the authoriseRequest for  {}", firstErrorMessage);
+            httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+            auditMessageService.auditMessage(new TokenMessagePair(dataRequest.getToken(), firstErrorMessage)).join();
+        } else {
```

Comment:
Ditto, could be `Optional.ofNullable(...).ifPresentOrElse(error -> {...}, () -> {...})`